xds/internal/clusterspecifier/rls/rls_test.go

```diff
@@ -75,19 +76,31 @@ func (s) TestParseClusterSpecifierConfig(t *testing.T) {
 		if (err != nil) != test.wantErr {
 			t.Fatalf("ParseClusterSpecifierConfig(%+v) returned err: %v, wantErr: %v", test.rlcs, err, test.wantErr)
 		}
-
+		if test.wantErr { // Successfully received an error.
+			return
+		}
 		// Marshal and then unmarshal into interface{} to get rid of
 		// nondeterministic protojson Marshaling.
 		lbCfgJSON, err := json.Marshal(lbCfg)
 		if err != nil {
 			t.Fatalf("json.Marshal(%+v) returned err %v", lbCfg, err)
 		}
+		var got interface{}
+		err = json.Unmarshal(lbCfgJSON, got)
+		if err != nil {
+			t.Fatalf("json.Unmarshal(%+v) returned err %v", lbCfgJSON, err)
+		}
 		wantCfgJSON, err := json.Marshal(test.wantConfig)
 		if err != nil {
 			t.Fatalf("json.Marshal(%+v) returned err %v", test.wantConfig, err)
 		}
-		if !reflect.DeepEqual(lbCfgJSON, wantCfgJSON) {
-			t.Fatalf("ParseClusterSpecifierConfig(%+v) = %v, want %v", test.rlcs, lbCfgJSON, wantCfgJSON)
+		var want interface{}
+		err = json.Unmarshal(wantCfgJSON, want)
+		if err != nil {
+			t.Fatalf("json.Unmarshal(%+v) returned err %v", lbCfgJSON, err)
+		}
+		if !cmp.Equal(want, got, cmpopts.EquateEmpty()) {
+			t.Fatalf("ParseClusterSpecifierConfig(%+v) returned expected, diff (-want +got):\\n%s", test.rlcs, cmp.Diff(want, got, cmpopts.EquateEmpty()))
 		}
```

Comment:
Use `if diff := cmp.Diff(want, got, cmpopts.EquateEmpty()); diff != "" {` and then there's no need to call `Diff` in the error message.